src/daft-scan/src/lib.rs

```diff
@@ -1,94 +1,284 @@
-use daft_dsl::Expr;
+use daft_dsl::{Expr, ExprRef};
 use daft_stats::{PartitionSpec, TableMetadata, TableStatistics};
+use file_format::FileFormatConfig;
 use serde::{Deserialize, Serialize};
 
 mod anonymous;
+pub mod file_format;
 mod glob;
+#[cfg(feature = "python")]
+pub mod py_object_serde;
+
 #[cfg(feature = "python")]
 pub mod python;
+pub mod storage_config;
 #[cfg(feature = "python")]
 pub use python::register_modules;
+use storage_config::StorageConfig;
 
-#[derive(Serialize, Deserialize, Clone, Copy, Debug)]
-pub enum FileType {
-    Parquet,
-    Avro,
-    Orc,
-    Csv,
-}
-
-impl FromStr for FileType {
-    type Err = DaftError;
-
-    fn from_str(file_type: &str) -> DaftResult<Self> {
-        use FileType::*;
-        if file_type.trim().eq_ignore_ascii_case("parquet") {
-            Ok(Parquet)
-        } else if file_type.trim().eq_ignore_ascii_case("avro") {
-            Ok(Avro)
-        } else if file_type.trim().eq_ignore_ascii_case("orc") {
-            Ok(Orc)
-        } else if file_type.trim().eq_ignore_ascii_case("csv") {
-            Ok(Csv)
-        } else {
-            Err(DaftError::TypeError(format!(
-                "FileType {} not supported!",
-                file_type
-            )))
-        }
-    }
-}
-
-#[derive(Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub enum DataFileSource {
     AnonymousDataFile {
-        file_type: FileType,
         path: String,
         metadata: Option<TableMetadata>,
         partition_spec: Option<PartitionSpec>,
         statistics: Option<TableStatistics>,
     },
     CatalogDataFile {
-        file_type: FileType,
         path: String,
         metadata: TableMetadata,
         partition_spec: PartitionSpec,
         statistics: Option<TableStatistics>,
     },
 }
 
-#[derive(Serialize, Deserialize)]
+impl DataFileSource {
+    pub fn get_path(&self) -> &str {
+        match self {
+            Self::AnonymousDataFile { path, .. } | Self::CatalogDataFile { path, .. } => path,
+        }
+    }
+    pub fn get_metadata(&self) -> Option<&TableMetadata> {
+        match self {
+            Self::AnonymousDataFile { metadata, .. } => metadata.as_ref(),
+            Self::CatalogDataFile { metadata, .. } => Some(metadata),
+        }
+    }
+
+    pub fn get_statistics(&self) -> Option<&TableStatistics> {
+        match self {
+            Self::AnonymousDataFile { statistics, .. }
+            | Self::CatalogDataFile { statistics, .. } => statistics.as_ref(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct ScanTask {
     // Micropartition will take this in as an input
-    source: DataFileSource,
-    columns: Option<Vec<String>>,
-    limit: Option<usize>,
+    pub source: DataFileSource,
+    pub file_format_config: Arc<FileFormatConfig>,
+    pub schema: SchemaRef,
+    pub storage_config: Arc<StorageConfig>,
+    // TODO(Clark): Directly use the Pushdowns struct as part of the ScanTask struct?
```

Comment:
@jaychia The TODO in the PR description "Consolidate filter/limit pushdowns to use the same Pushdown struct." was also referring to this, where we could use the `Pushdowns` struct as a `pushdowns` field on the `ScanTask`.